### PR TITLE
Fix documentation and some miscellaneous things

### PR DESCRIPTION
**Pull request by molereddy**
# What does this PR do?
documentation
- adds more stuff in README: better badges, change arxiv to ICLR citation for MUSE
- clarifies deepspeed usage / multi-gpu eval issue in docs
- documents more hydra features
- does many miscellaneous reformattings and cleanups to documentation

misc
- adds a on-push workflow
- adds experiment example config files that reflect latest code
- better handles lack of flash-attn support while loading model
- updates logging to clarify deepspeed usage / multi-gpu eval issue

Fixes # (issue)
#94

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Have you gone through the contributions [guide](../docs/contributing.md)?
- [x] Are your changes documented? Read documentation guidelines [here](../README.md#-further-documentation).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,14 +5,23 @@
...
       <a href="https://huggingface.co/open-unlearning">
-        <img src="https://img.shields.io/badge/Hugging%20Face-white?logo=huggingface" alt="Hugging Face">
+        <img src="https://img.shields.io/badge/%F0%9F%A4%97%20Hugging%20Face-Model-blue" alt="HuggingFace 🤗">
       </a>
       <a href="https://github.com/locuslab/open-unlearning">
-        <img src="https://img.shields.io/github/stars/locuslab/open-unlearning?style=social" alt="GitHub Repo stars">
+        <img src="https://img.shields.io/github/repo-size/locuslab/open-unlearning" alt="GitHub repo size">
+      </a>
+      <a href="https://github.com/locuslab/open-unlearning">
+        <img src="https://img.shields.io/github/languages/top/locuslab/open-unlearning" alt="GitHub top language">
```
Comment (Dornavineeth) [inline README.md:21]: Not rendering properly

File: README.md [hunk 2]
```diff
@@ -5,14 +5,23 @@
...
+      </a>
+      <a href="https://github.com/locuslab/open-unlearning/blob/main/LICENSE">
+        <img src="https://img.shields.io/badge/License-MIT-blue" alt="License: MIT">
```
Comment (Dornavineeth) [inline README.md]: There is some blue line between each icon. Shall we try to remove it?

File: configs/experiment/examples/muse_unlearn.yaml [hunk 3]
```diff
@@ -22,13 +22,15 @@ trainer:
     per_device_train_batch_size: 4
     per_device_eval_batch_size: 16
     gradient_accumulation_steps: 8
-    learning_rate: 1.0e-05
+    learning_rate: 3.0e-05
```
Comment (Dornavineeth) [inline configs/experiment/examples/muse_unlearn.yaml:25]: Any reason for changing this?

File: configs/experiment/examples/tofu_eval.yaml [hunk 4]
```diff
@@ -1,22 +1,90 @@
...
   model_args:
-    device_map: auto
-    pretrained_model_name_or_path: locuslab/tofu_ft_llama2-7b
+    device_map: cuda
+    pretrained_model_name_or_path: open-unlearning/tofu_Llama-3.2-1B-Instruct_full
     attn_implementation: flash_attention_2
     torch_dtype: bfloat16
   tokenizer_args:
-    pretrained_model_name_or_path: locuslab/tofu_ft_llama2-7b
+    pretrained_model_name_or_path: meta-llama/Llama-3.2-1B-Instruct
```
Comment (Dornavineeth) [inline configs/experiment/examples/tofu_eval.yaml:8]: better change this to ope-unlearning full finetuned model.

File: configs/experiment/examples/tofu_eval.yaml [hunk 5]
```diff
@@ -1,22 +1,90 @@
...
+
+
+      '
+    asst_end_tag: <|eot_id|>
 mode: eval
-task_name: eval
+task_name: SAMPLE_EVAL
+seed: 0
 eval:
   tofu:
```
Comment (Dornavineeth) [inline configs/experiment/examples/tofu_eval.yaml:30]: Can we have just one metric here?
Reason we are adding this file is to help the user to get a sense of config structure quickly.
Adding more metrics is not helping much.

**Review by Dornavineeth — CHANGES_REQUESTED** [review]
Looks good. Just a few things to check.

File: README.md (hunk 2 above)
Comment (Dornavineeth) [reply README.md]: We should also fix the bug of test - no status icon. Would be great if you do this in this PR itself.

File: configs/experiment/examples/muse_unlearn.yaml (hunk 3 above)
Comment (molereddy) [reply configs/experiment/examples/muse_unlearn.yaml:25]: I just dumped it from some random experiment. I don't think this matters.

File: configs/experiment/examples/tofu_eval.yaml (hunk 5 above)
Comment (molereddy) [reply configs/experiment/examples/tofu_eval.yaml:30]: Done!

File: README.md (hunk 2 above)
Comment (molereddy) [reply README.md]: I think the test no status is because the workflow config mentions tests only in PR. There is no cfg for normal branch.

Comment (molereddy) [reply README.md]: Done! 
- Pushed changes fixing this in PR. But issue is you have to set the correct repo path and branch name. So when I revert to locuslab and branch main tests will be no status again before merge. 
- Another issue, if people are working in a fork or a branch they will not see status of their own tests but that of locuslab openunlearning. Hope this is ok!

Comment (molereddy) [reply README.md]: Fixed blue line. Everything pretty now!

Comment (molereddy) [reply README.md]: Once you approve I will revert the changes to tests remote repo path and set back to locuslab main

**Review by Dornavineeth — APPROVED** [review]
